src/Form/AuthenticationForm.php

```diff
@@ -118,35 +118,148 @@ public function buildForm(array $form, FormStateInterface $form_state) {
 
-    $options = $this->keyRepository->getKeyNamesAsOptions(['type_group' => 'apigee_edge']);
-    if (empty($options)) {
-      $this->messenger->addWarning(t('There is no available key for connecting to Apigee Edge server. <a href=":link">Create a new key.</a>', [
-        ':link' => Url::fromRoute('entity.key.add_form', ['destination' => 'admin/config/apigee-edge/settings'])->toString(),
-      ]));
+    // Loading basic authentication keys.
+    $basic_auth_keys = $this->keyRepository->getKeyNamesAsOptions(['type' => 'apigee_edge_basic_auth']);
+    foreach ($basic_auth_keys as $key_id => $key_name) {
+      $basic_auth_keys[$key_id] = $this->t('@key_name <a href=":url">Edit</a>', [
+        '@key_name' => $key_name,
+        ':url' => Url::fromRoute('entity.key.edit_form', ['key' => $key_id, 'destination' => 'admin/config/apigee-edge/settings'])->toString(),
+      ]);
     }
-    foreach ($options as $key_id => $key_name) {
-      $options[$key_id] = $this->t('@key_name <a href=":url">Edit</a>', [
+    $basic_auth_default_value = array_key_exists($config->get('active_key'), $basic_auth_keys) ? $config->get('active_key') : NULL;
+
+    // Loading OAuth keys.
+    $oauth_keys = $this->keyRepository->getKeyNamesAsOptions(['type' => 'apigee_edge_oauth']);
+    foreach ($oauth_keys as $key_id => $key_name) {
+      $oauth_keys[$key_id] = $this->t('@key_name <a href=":url">Edit</a>', [
         '@key_name' => $key_name,
         ':url' => Url::fromRoute('entity.key.edit_form', ['key' => $key_id, 'destination' => 'admin/config/apigee-edge/settings'])->toString(),
       ]);
     }
-    $default_value = array_key_exists($config->get('active_key'), $options) ? $config->get('active_key') : NULL;
+    $oauth_default_value = array_key_exists($config->get('active_key'), $oauth_keys) ? $config->get('active_key') : NULL;
 
-    $form['authentication']['key'] = [
-      '#type' => 'radios',
-      '#title' => t('Keys'),
-      '#options' => $options,
-      '#access' => !empty($options),
-      '#default_value' => $default_value,
+    // Loading OAuth token keys.
+    $oauth_token_keys = $this->keyRepository->getKeyNamesAsOptions(['type' => 'apigee_edge_oauth_token']);
+    foreach ($oauth_token_keys as $key_id => $key_name) {
+      $oauth_token_keys[$key_id] = $this->t('@key_name <a href=":url">Edit</a>', [
+        '@key_name' => $key_name,
+        ':url' => Url::fromRoute('entity.key.edit_form', ['key' => $key_id, 'destination' => 'admin/config/apigee-edge/settings'])->toString(),
+      ]);
+    }
+    $oauth_token_default_value = array_key_exists($config->get('active_key_oauth_token'), $oauth_token_keys) ? $config->get('active_key_oauth_token') : NULL;
+
+    $form['authentication']['key_type'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Apigee Edge API authentication type'),
+      '#options' => [
+        'apigee_edge_basic_auth' => $this->t('Basic authentication'),
+        'apigee_edge_oauth' => $this->t('OAuth (SAML)'),
```

Comment:
IIRC we have not got a clear answer yet whether we should remove "SAML" from the strings, we can do it in an upcoming sprint.

Reply:
Agreed.